File: javascript/clock/clock.js
var Clock = function (hour, minutes) {
  this.setMinutes = function () {
    while (this.minutes >= 60) {
      this.minutes = this.minutes - 60;
      this.hour = this.hour + 1;
    }
    while (this.minutes < 0) {
      this.minutes += 60;
      this.hour -= 1;
    }
  };

  this.setHours = function () {
    while (this.hour >= 24) {
      this.hour = this.hour - 24;
    }
    while (this.hour < 0) {
      this.hour += 24;
    }
  };

  this.plus = function(minutes) {
    this.minutes += minutes;
    this.setMinutes();
    this.setHours();
    return this;
  };

  this.minus = function (minutes) {
    this.minutes -= minutes;
    this.setMinutes();
    this.setHours();
    return this;
  };

  this.hour = hour;
  this.minutes = minutes != undefined ? minutes : 0;
  this.setMinutes();
  this.setHours();

  this.toString = function () {
    return ('0' + this.hour.toString()).slice(-2) + ":" + ('0' + this.minutes.toString()).slice(-2);
  };

  this.equals = function (other) {
    return this.hour == other.hour && this.minutes == other.minutes;
  }
};

module.exports.at = function(hour, minutes) {
  return new Clock(hour, minutes);
};

File: csharp/clock/Clock.cs
class Clock
{
  public Clock(int aHours = 0, int aMinutes = 0)
  {
    hours = aHours;
    if (hours >= 24)
      hours -= 24;
    if (aMinutes == 60)
    {
      hours++;
      minutes = 0;
    }
    else
      minutes = aMinutes;
  }

  public Clock Add(int aMinutes)
  {
    var newHours = hours;
    var newMinutes = minutes + aMinutes;
    while (newMinutes > 60)
    {
      newHours++;
      if (newHours > 23)
        newHours = 0;
      newMinutes -= 60;
    }
    return new Clock(newHours, newMinutes);
  }

  public Clock Subtract(int aMinutes)
  {
    var newHours = hours;
    var newMinutes = minutes - aMinutes;
    while (newMinutes < 0)
    {
      newHours--;
      if (newHours < 0)
        newHours = 23;
      newMinutes += 60;
    }
    return new Clock(newHours, newMinutes);
  }

  public override string ToString()
  {
    return hours.ToString().PadLeft(2, '0') + ":" + minutes.ToString().PadLeft(2, '0');
  }

  public override bool Equals(object obj)
  {
    return obj.ToString() == ToString();
  }

  private int hours;
  private int minutes;
}